src/gfxinit.cpp

```diff
@@ -481,18 +481,27 @@ template <class Tbase_set>
 	if (BaseMedia<Tbase_set>::used_set != nullptr) return true;
 
 	const Tbase_set *best = nullptr;
+
+	auto IsBetter = [&best] (const auto *current) {
```

Comment:
How about turning this into a comparator function and using std::sort ?

Reply:
It's not a STL collection. `std::sort` might be a bit heavy for what a `std::ranges::max_element` could do.
Changing the collection to a STL collection is in my opinion beyond the scope of this PR.